src/pages/DatasetStatistics.jsx

```diff
@@ -20,8 +20,18 @@ export default function DatasetStatistics(props) {
   const [isLoading, setIsLoading] = useState(true);
 
   const applyForAccess = async () => {
-    const darDraft = await DAR.postDarDraft({ datasetId: [datasetId]  });
-    history.push(`/dar_application/${darDraft.referenceId}`);
+    try {
+      const draftResponse = await DAR.postDarDraft({ datasetId: [datasetId]  });
+      if (draftResponse.referenceId) {
+        history.push(`/dar_application/${draftResponse.referenceId}`);
+      } else if (draftResponse.message) {
+        Notifications.showError({ text: draftResponse.message + ' Please contact customer support for help.' });
+      } else {
+        Notifications.showError({ text: 'Error: Unable to create a Draft Data Access Request' });
```

Comment:
Possible improvement, `throw` here so it bubbles to the `catch` statement.

Reply:
I am seeing [this warning](https://www.jetbrains.com.cn/en-us/help/inspectopedia/ExceptionCaughtLocallyJS.html) with this suggestion. Another approach I would like to take is to extract the Apply functionality to a separate component - if we do tackle that, maybe we can improve the control flow there.